### PR TITLE
fix(http-parser): Correctly decode + as space in URL components

### PR DESCRIPTION
**Pull request by jneums**
**Problem:**
The query string parser was failing to correctly handle space characters encoded as `+` (e.g., `scope=openid+profile`). This is a standard encoding method for `application/x-www-form-urlencoded` data, but our parser was only handling percent-encoding (`%20`). This resulted in scope strings being incorrectly processed as `"openid+profile"` instead of the correct `"openid profile"`.

**Root Cause:**
The logic for handling `+` as a space was missing from the main query string parsing loop. The `decodeURIComponent` utility correctly handles percent-encoding, but the `+` to space conversion is a distinct step specific to the `application/x-www-form-urlencoded` format.

**Solution:**
The query string parsing logic in `HttpParser.mo` has been updated to first replace all `+` characters with spaces, and *then* pass the result to the `decodeURIComponent` function.

This change correctly separates the two decoding concerns:
1.  **Form-URL-Encoded Parsing:** The main loop now correctly handles the `+` to space conversion.
2.  **URI Component Decoding:** The `Utils.decodeURIComponent` function remains focused on its single responsibility of handling percent-encoded characters.

This results in a more robust and architecturally sound parser.

**Validation:**
Added a new unit test to `test_http_parser.mo` within the `SearchParams Tests` suite. This test specifically asserts that a query string like `scope=openid+profile+prometheus:charge` is correctly decoded to `"openid profile prometheus:charge"`. This test now passes, confirming the fix and preventing future regressions.

### CODE REVIEW

File: src/Utils.mo [hunk 1]
```diff
@@ -182,27 +182,62 @@ module {
...
+        let decodedBuffer = Buffer.Buffer<Nat8>(sourceBytes.size());
+        var i = 0;
+
+        label parseBytes while (i < sourceBytes.size()) {
+            let byte = sourceBytes[i];
+
+            // Compare the byte directly with the Nat8 value for '%', which is 37.
+            if (byte == (37 : Nat8)) {
+                // Check if there are at least two characters to look ahead.
+                if (i + 2 < sourceBytes.size()) {
```
[the remark this answers is not in the document]
Comment (jneums) [reply src/Utils.mo:211]: Copilot is incorrect here. Added a test for the edge case.

File: src/Utils.mo [hunk 2]
```diff
@@ -182,27 +182,62 @@ module {
...
+                        let hexString = Text.fromChar(char1) # Text.fromChar(char2);
+                        switch (Hex.decode(hexString)) {
+                            case (#ok(decodedByteBlob)) {
+                                if (decodedByteBlob.size() == 1) {
+                                    decodedBuffer.add(decodedByteBlob[0]);
+                                    i += 3; // Advance index past the '%' and the two hex digits.
+                                    continue parseBytes;
+                                };
+                            };
+                            case (#err(_)) { /* Unreachable */ };
```
[the remark this answers is not in the document]
Comment (jneums) [reply src/Utils.mo]: Improved the checks and added tests for this case.

**Comment by tomijaga** [comment]
Hey @jneums,
Thanks for adding support for decoding URLs that contains `+` characters.
I'm going to merge it your changes and push the changes to mops

**Comment by tomijaga** [comment]
@jneums, the latest version `v0.4.1` is now uploaded and available on mops: https://mops.one/http-parser@0.4.1/versions

**Comment by jneums** [comment]
> @jneums, the latest version `v0.4.1` is now uploaded and available on mops: https://mops.one/http-parser@0.4.1/versions

Thanks @tomijaga! I'll bump my dependencies asap.
